nomad/state/state_store.go

```diff
@@ -859,8 +859,14 @@ func (s *StateStore) UpdateAllocsFromClient(index uint64, allocs []*structs.Allo
 
 	// Handle each of the updated allocations
 	for _, alloc := range allocs {
-		if err := s.updateSummaryWithAlloc(index, alloc, watcher, txn); err != nil {
-			return fmt.Errorf("error updating job summary: %v", err)
+		rawJob, err := txn.First("jobs", "id", alloc.JobID)
```

Comment:
@dadgar That would be outside of the current transaction. I think it might be best to not retrieve anything from the state store using a new transaction when we already have a write  transaction open.